runtime/src/iree/builtins/ukernel/arch/x86_64/pack_x86_64_avx512_base.c

```diff
@@ -19,8 +19,8 @@ void iree_uk_pack_tile_16x16_x32_x86_64_avx512_base_direct(
   const iree_uk_int8_t* IREE_UK_RESTRICT in_ptr = in_tile_ptr;
   iree_uk_int8_t* IREE_UK_RESTRICT out_ptr = out_tile_ptr;
   for (; outer_size1 > 0; --outer_size1) {
-    iree_uk_copy_8x32xi8_strided_to_strided(out_ptr, in_ptr, 64,
-                                            4 * in_stride0);
+    iree_uk_copy_16x64xi8_strided_to_strided(out_ptr, in_ptr, 64,
```

Comment:
Noob question: do we have a decoder for the conventions? (naming, restrictions on strides and sizes for input, when this can be called)

Reply:
`iree_uk_pack_tile_16x16_x32_x86_64_avx512_base_direct` is a tile function. Each ukernel has some generic, shared outer loops written in C, shared among all cases (we are trying to minimize code size), and then the case-specific, architecture-specific hot inner loop is provided by a tile function like this. Here is the whole implementation of the pack ukernel:
https://github.com/openxla/iree/blob/6dcca4094988af88035fe72159df62753c790b59/runtime/src/iree/builtins/ukernel/pack.c#L230-L238

So `iree_uk_pack_tile_16x16_x32_x86_64_avx512_base_direct` is what is being pointed to by the `tile_func` function pointer here. The naming convention for `pack` tile functions is as follows:

```
iree_uk_pack_tile_${tile_shape}_${size_only_element_type}_${arch}_${arch_extra_features}_${direct_or_transpose}
```

where:
* `tile_shape` is the shape of tiles that data is getting tiled into. Here 16x16 elements.
* `size_only_element_type` is e.g. `x32` for "any 32-bit element type". For now, `pack` ukernels don't care about actual element type, just size. The prefix `x` means that. If we cared about more than size we would start naming tile functions with `f32` or `i32` instead.
* `direct_or_transpose` refers to whether the packing performs transposition of each tile or not. This is used for the RHS of matmuls only.

Now, `iree_uk_copy_16x64xi8_strided_to_strided` is just some internal implementation function.  You see here this code saying "packing this 16x16 tile of 32bit elements is the same as packing 16x64 8bit elements so let's canonicalize all to such 8bit element helpers to remove an extraneous degree of freedom".